rtc_data_service/tests/ecalls/issue_execution_token.rs

```diff
@@ -0,0 +1,90 @@
+use std::{convert::TryInto, str::FromStr};
+
+use rtc_types::ExecReqMetadata;
+use serde::{Deserialize, Serialize};
+use sgx_types::sgx_target_info_t;
+
+use crate::{helpers, CRYPTO_BOX_BOXZEROBYTES, CRYPTO_BOX_ZEROBYTES};
+
+#[derive(Serialize, Deserialize)]
+pub struct ExecReqData {
+    dataset_uuid: [u8; 16],
+    dataset_access_key: [u8; 24],
+    exec_module_hash: [u8; 32],
+    number_of_uses: u32,
+}
+
+#[test]
+fn test_issue_execution_token_success() {
+    let enclave = helpers::init_auth_enclave();
+
+    let enclave_pubkey = enclave
+        .create_report(&sgx_target_info_t::default())
+        .unwrap()
+        .enclave_held_data;
+
+    let mut pubkey = [0_u8; 32];
+    let mut privkey = [0_u8; 32];
+
+    sodalite::box_keypair_seed(&mut pubkey, &mut privkey, &[2_u8; 32]);
+
+    let uuid = uuid::Uuid::from_str("dd12012195c04ae8990ebd2512ae03ab").unwrap();
+    let exec_module_hash: Vec<u8> = (0u8..32).collect();
+
+    let req_json = serde_json::to_vec(&ExecReqData {
+        dataset_uuid: *uuid.as_bytes(),
+        dataset_access_key: [1; 24],
+        exec_module_hash: exec_module_hash.try_into().unwrap(),
+        number_of_uses: 10,
+    })
+    .unwrap();
+
+    let plaintext = [vec![0_u8; 32], req_json].concat();
+    let mut ciphertext = vec![0_u8; plaintext.len()];
+    let nonce = [8_u8; 24];
+
+    sodalite::box_(
+        &mut ciphertext,
+        &plaintext,
+        &nonce,
+        &enclave_pubkey,
+        &privkey,
+    )
+    .unwrap();
+
+    let result = enclave
+        .issue_execution_token(
+            &ciphertext[CRYPTO_BOX_BOXZEROBYTES..],
+            ExecReqMetadata {
+                uploader_pub_key: pubkey,
+                nonce,
+            },
+        )
+        .unwrap();
```

Comment:
This test fails here. Are you aware?

Reply:
It passes here.

@Nghondzweni: Was this maybe because of a stale enclave library?